ReportUtilityBills/tests/utility_bills_test.rb

```diff
@@ -34,233 +35,296 @@ class ReportUtilityBillsTest < MiniTest::Test
     actual_bills = _get_actual_bills(bills_csv)
-    assert_operator(actual_bills['Coal: Total ($)'], :>, 0)
+    expected_val = actual_bills['Test 1: Coal: Total ($)']
+    assert_in_delta(expected_val * 2, actual_bills['Test 2: Coal: Total ($)'], 1)
+    assert_in_delta(expected_val / 2, actual_bills['Test 3: Coal: Total ($)'], 1)
   end
 
   def test_workflow_leap_year
-    @args_hash['hpxml_path'] = '../workflow/sample_files/base-location-AMY-2012.xml'
+    @args_hash['hpxml_path'] = File.absolute_path(@tmp_hpxml_path)
+    hpxml = HPXML.new(hpxml_path: File.join(@sample_files_path, 'base-location-AMY-2012.xml'))
+    hpxml.header.utility_bill_scenarios.add(name: 'Test')
+    XMLHelper.write_file(hpxml.to_oga, @tmp_hpxml_path)
     bills_csv = _test_measure()
     assert(File.exist?(bills_csv))
     actual_bills = _get_actual_bills(bills_csv)
-    assert_operator(actual_bills['Total ($)'], :>, 0)
+    assert_operator(actual_bills['Test: Total ($)'], :>, 0)
   end
 
   def test_workflow_semi_annual_run_period
-    @args_hash['hpxml_path'] = '../workflow/sample_files/base-simcontrol-runperiod-1-month.xml'
+    @args_hash['hpxml_path'] = File.absolute_path(@tmp_hpxml_path)
+    hpxml = HPXML.new(hpxml_path: File.join(@sample_files_path, 'base-simcontrol-runperiod-1-month.xml'))
+    hpxml.header.utility_bill_scenarios.add(name: 'Test')
+    XMLHelper.write_file(hpxml.to_oga, @tmp_hpxml_path)
     bills_csv = _test_measure()
     assert(File.exist?(bills_csv))
     actual_bills = _get_actual_bills(bills_csv)
-    assert_operator(actual_bills['Total ($)'], :>, 0)
+    assert_operator(actual_bills['Test: Total ($)'], :>, 0)
+  end
+
+  def test_workflow_no_bill_scenarios
+    @args_hash['hpxml_path'] = File.absolute_path(@tmp_hpxml_path)
+    hpxml = HPXML.new(hpxml_path: File.join(@sample_files_path, 'base.xml'))
+    XMLHelper.write_file(hpxml.to_oga, @tmp_hpxml_path)
+    bills_csv = _test_measure()
+    assert(!File.exist?(bills_csv))
   end
 
   def test_auto_marginal_rate
-    fuel_types = [FT::Elec, FT::Gas, FT::Oil, FT::Propane]
+    fuel_types = [HPXML::FuelTypeElectricity, HPXML::FuelTypeNaturalGas, HPXML::FuelTypeOil, HPXML::FuelTypePropane]
 
     # Check that we can successfully look up "auto" rates for every state
     # and every fuel type.
-    runner = OpenStudio::Measure::OSRunner.new(OpenStudio::WorkflowJSON.new)
     Constants.StateCodesMap.keys.each do |state_code|
       fuel_types.each do |fuel_type|
-        flatratebuy = @measure.get_auto_marginal_rate(runner, state_code, fuel_type, 0)
+        flatratebuy = UtilityBills.get_auto_marginal_rate(state_code, fuel_type, 0)
         refute_nil(flatratebuy)
       end
     end
 
     # Check that any other state code is gracefully handled (no error)
     fuel_types.each do |fuel_type|
-      @measure.get_auto_marginal_rate(runner, 'XX', fuel_type, 0)
+      UtilityBills.get_auto_marginal_rate('XX', fuel_type, 0)
     end
   end
 
   def test_warning_region
+    skip
```

Comment:
Remove test altogether?

Reply:
Removed.

Reply:
Sorry, I missed this. Why does this need to be skipped/removed?

Reply:
We don't pass `runner` into `hpxml_defaults.rb`.

Reply:
You can start passing it in. It's not currently passed in because it wasn't needed before.